frontend/src/views/AddAnimals/AnimalFormComponents/DairyCattle.tsx

```diff
@@ -200,7 +216,7 @@ export default function DairyCattle({
           name="animalsPerFarm"
           value={formData?.animalsPerFarm?.toString()}
           onChange={(e: string) => {
-            handleInputChanges({ animalsPerFarm: e });
+            handleInputChanges({ animalsPerFarm: parseFloat(e) });
```

Comment:
You should do Number(e) instead. A type "number" TextField doesn't have non-numerical chars to worry about.

Reply:
Fixed